src/toolchain.rs

```diff
@@ -408,16 +408,21 @@ impl<'a> Toolchain<'a> {
         buf
     }
 
-    pub fn doc_path(&self, relative: impl AsRef<Path>) -> anyhow::Result<PathBuf> {
+    pub fn doc_path(&self, maybe_relative: impl AsRef<Path>) -> anyhow::Result<PathBuf> {
```

Comment:
Personally, if I were to read `fn doc_path(&self, path: impl AsRef<Path>)`, I would not have known what to pass as `path`. `maybe_relative` is definitely a long name and I kind of dislike it. However I don't think `path` is a better option.

Reply:
I think it's better to just call it `relative`, since implicitly we still have absolute + absolute = absolute, which is also the convention for e.g. `std`.

Reply:
On that note, could we just store `self.html_root = doc_dir.extend(["share", "doc", "rust", "html"]);` or so then `doc_path` is just `self.html_root.join(relative)`.

Reply:
@djc Hmmm maybe you have marked this conversation as resolved by mistake? Anyway, I think what @ChrisDenton has pointed out is a pre-existing issue, and looking at the current source I'm not sure what the best thing to do currently is.

We could cache the extended base path somewhere, but there's nothing stopping us from changing the `Toolchain` instance mid-way so the cache should be invalidated...